prow/repoowners/repoowners.go

```diff
@@ -794,7 +794,7 @@ func (o *RepoOwners) entriesForFile(path string, people map[string]map[*regexp.R
 	if !o.enableMDYAML || !strings.HasSuffix(path, ".md") {
 		// if path is a directory, this will remove the leaf directory, and returns "." for topmost dir
 		d = filepath.Dir(d)
-		d = canonicalize(path)
```

Comment:
This changes the result, are you sure this is correct?

Reply:
Not 100%. The value for d from 
```
d = filepath.Dir(d)
```
Is never used as it is overwritten by
```
d = canonicalize(path)
```
And is hence redundant. There is some code in the same file which does 
```
                d = filepath.Dir(d)
		d = canonicalize(d)
```
So I assume this should be the same...

Reply:
Can you instead change it to just `d = canonicalize(d)` (without the `filepath.Dir(d)`)? That would preserve the current behavior. Unfortunately we don't have tests for this it seems

Reply:
Sure, no problems. I've updated and removed the comment as it refers to ```filepath.Dir(d)```.